src/components/FormType.tsx

```diff
@@ -50,6 +50,17 @@ const GitHubFormType = ({ modals }: GitHubFormTypeProps) => {
   const createAutomation = api.automation.createAutomation.useMutation();
   const userAutomations = api.automation.getUserAutomations.useQuery();
 
+  const [owner, setOwner] = useState("");
+  const [repositoryEventSelected, setRepositoryEventSelected] = useState(false);
+  const [repositoryName, setRepositoryName] = useState("");
+  const [ifCondition, setIfCondition] = useState<
+    "issues" | "pull_request" | "push" | "star" | ""
```

Comment:
defining the possible values in one place makes it much easier to extend later on